qiskit/providers/aer/backends/cluster/clusterjobset.py

```diff
@@ -0,0 +1,236 @@
+        the timeout is reached.
+
+        Args:
+           timeout: Number of seconds to wait for job results.
+           raises: whether or not to re-raise exceptions thrown by jobs
+
+        Returns:
+            qiskit.Result: Result object
+
+        Raises:
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+
+        """
+        return self.results(timeout, raises).combine_results()
+
+    @requires_submit
+    def results(
+            self,
+            timeout: Optional[float] = None,
+            raises: Optional[bool] = False
+    ) -> CResults:
+        """Return the results of the jobs.
+
+        This call will block until all job results become available or
+        the timeout is reached. Analogous to dask.client.gather()
+
+        Args:
+           timeout: Number of seconds to wait for job results.
+           raises: whether or not to re-raise exceptions thrown by jobs
+
+        Returns:
+            A :class:`CResults`
+            instance that can be used to retrieve results
+            for individual experiments.
+
+        Raises:
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+        """
+        if self._results is not None:
+            return self._results
+
+        start_time = time.time()
+        original_timeout = timeout
+
+        # We'd like to use futures.as_completed or futures.wait
+        #   however this excludes the use of dask as executor
+        #   because dask's futures are not ~exactly~ the same.
+        res = []
+        for cjob in self._futures:
+            try:
+                result = cjob.result(timeout=timeout, raises=raises)
+                if result is None or not result.success:
+                    if result:
+                        logger.warning('ClusterJob %s Error: %s', cjob.name(), result.header)
+                    else:
+                        logger.warning('ClusterJob %s did not return a result', cjob.name())
+                res.append(result)
+            except AerClusterTimeoutError:
+                raise AerClusterTimeoutError(
+                    'Timeout while waiting for the results of experiment {}'.format(
+                        cjob.name()))
+            if timeout:
+                timeout = original_timeout - (time.time() - start_time)
+                if timeout <= 0:
+                    raise AerClusterTimeoutError(
+                        "Timeout while waiting for JobSet results")
+
+        self._results = CResults(self, res)
+
+        return self._results
+
+    @requires_submit
+    def cancel(self) -> None:
+        """Cancel all jobs in this job set."""
+        for cjob in self._futures:
+            cjob.cancel()
+
+    @requires_submit
+    def job(self, experiment: Union[str, QuantumCircuit, Schedule]) -> Tuple[CJob, int]:
```

Comment:
Maybe this should be broken up into two methods:
```python
def worker(self, experiment: Union[Integral, str, QuantumCircuit, Schedule]) -> Union[int, List[int]]
```
where you can also pass in the experiment by integer position from the original execution list. The return type could also include a list of integers to allow for spreading a single circuit into multiple workers in the future (eg for shot/noise parallelization).

Then the job method could be
```python
    def worker_job(self, worker: Union[int, Iterable[int]]) -> Union[AerJob, List[AerJob]]:
```